### PR TITLE
Users/nigurr/port xplat agent from master

### PR DESCRIPTION
**Pull request by nigurr**
Porting changes from master to Releases/0.5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 A cross platform build agent for Microsoft Visual Studio Team Services (VSTS) and Team Foundation Server (TFS).  Supported on Mac OSX and Linux.
 
+> NOTICE: This agent will soon be replaced with a single open sourced cross platform agent for Windows, OSX and Linux written in CoreCLR.  This agent is still supported but many of the limitations and pending features for this agent are being written into the new agent from the start.  [See more here](https://github.com/Microsoft/vsts-agent/blob/master/README.md).  We will triage every issue and ensure it's in the new agent or moved over.
```
Comment (acesiddhu) [inline README.md]: get these reviewed by Bryan

File: src/agent/context.ts [hunk 2]
```diff
@@ -150,20 +150,16 @@ export class HostContext extends Context implements cm.IHostContext {
...
-        this.config = config;
-        this.workFolder = cm.getWorkPath(config);
-        
-        ensureTrace(this);
-        
-        this._fileWriter = fileWriter;
-        
-        var writers: cm.IDiagnosticWriter[] = [this._fileWriter];
-        
+        var writers: cm.IDiagnosticWriter[] = [fileWriter];
```
Comment (acesiddhu) [inline src/agent/context.ts]: why are these needed?

**Comment by AshwiniChalla** [comment]
approved for test publish changes. get approved from respective owners for other changes.

**Comment by bryanmacfarlane** [comment]
We shouldn't take this.  I will create releases/0.6 based off of master today
